### PR TITLE
fix: Requested by @Aprillion  #54

### PR DESCRIPTION
**Pull request by AhmedEldessouki**

### CODE REVIEW

**Comment by AhmedEldessouki** [comment]
resolves #54

**Comment by AhmedEldessouki** [comment]
```
husky > pre-commit (node v14.15.4)
[STARTED] Preparing...
[SUCCESS] Preparing...
[STARTED] Running tasks...
[STARTED] Running tasks for *.+(js|jsx|ts|tsx)
[STARTED] Running tasks for *.+(json|yml|yaml|css)
[STARTED] yarn lint:precomit
[SUCCESS] prettier --write
[SUCCESS] Running tasks for *.+(json|yml|yaml|css)
[SUCCESS] yarn lint:precomit
[SUCCESS] Running tasks for *.+(js|jsx|ts|tsx)
[SUCCESS] Running tasks...
[STARTED] Applying modifications...
[SUCCESS] Applying modifications...
[STARTED] Cleaning up...
[SUCCESS] Cleaning up...
```

**Comment by AhmedEldessouki** [comment]
@JacobMGEvans should i keep it that (with the new script) or remove the new script?

File: lint-staged.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  '*.+(js|jsx|ts|tsx)': ['yarn lint'],
-  // '*.+(js|jsx|ts|tsx)': ['eslint', 'yarn test:precommit'],
-  '*.+(js|jsx|json|yml|yaml|css|ts|tsx)': ['prettier --write'],
+  '*.+(js|jsx|ts|tsx)': ['yarn lint:precomit'],
+  // '*.+(js|jsx|ts|tsx)': ['yarn lP:precommit', 'yarn test:precommit'],
```
Comment (JacobMGEvans) [inline lint-staged.config.js:3]:
```suggestion
```

**Review by JacobMGEvans — CHANGES_REQUESTED** [review]

**Comment by merodiro** [comment]
This would revert the changes I made to speed pre-commit hook in #56

also there is no need to run both `yarn lint` and `yarn format` because `eslint` will run `prettier` with this `plugin:prettier/recommended` in the config

**Comment by AhmedEldessouki** [comment]
> 
> 
> This would revert the changes I made to speed pre-commit hook in #56
> 
> also there is no need to run both `yarn lint` and `yarn format` because `eslint` will run `prettier` with this `plugin:prettier/recommended` in the config

interesting i didnt know that. ok i will do the changes

**Review by merodiro — COMMENTED** [review]
Running `yarn lint` is slower than running `eslint --fix` because it would run eslint on every file in the repo and not just the changed files. you can check #56 for the numbers

**Comment by AhmedEldessouki** [comment]
@JacobMGEvans @merodiro 
 one of the CI `windows latest, 14` failed at installing dependencies (firebase). and that's what i am using and running yarn doesn't give any errors. what shall i do?

**Review by JacobMGEvans — APPROVED** [review]

**Pull request merged**